.github/workflows/terraform-observe_scheduler.yaml

```diff
@@ -49,8 +49,38 @@ jobs:
       - uses: hashicorp/setup-terraform@v2
         with:
           terraform_version: ${{ inputs.terraform-version }}
+
+      - name: Checkout Main
+        uses: actions/checkout@v3
+        if: startsWith(${{ matrix.directory }}, 'upgrade')
```

Comment:
Upgrade should be a mode of running existing test modules, versus requiring a new test module.

Reply:
will add a per module config - set upgrade to true